### ISSUE TITLE
Multiple ingress "CREATE" events investigation

### ISSUE THREAD

**Issue by jasonBirchall** [ministryofjustice/cloud-platform#2389]
https://mojdt.slack.com/archives/C57UPMZLY/p1601472980119200

We're seeing a large number of ingress rule creations over the past few days. We're unsure what is causing this but know a large portion of them either cloud-platform integration tests or LAA-CLA-backend-staging/uat.

**Comment by vijay-veeranki** [comment]
```
https://kibana.cloud-platform.service.justice.gov.uk/_plugin/kibana/app/kibana#/discover?_g=(filters:!(),refreshInterval:(pause:!t,value:0),time:(from:'2020-09-29T16:00:00.000Z',to:'2020-09-29T17:00:00.000Z'))&_a=(columns:!(_source),filters:!(('$state':(store:appState),meta:(alias:!n,disabled:!f,index:d2f00a00-fbf4-11ea-86fe-a5f65a6b0b79,key:log_processed.event.involvedObject.kind,negate:!f,params:(query:Ingress),type:phrase,value:Ingress),query:(match:(log_processed.event.involvedObject.kind:(query:Ingress,type:phrase)))),('$state':(store:appState),meta:(alias:!n,disabled:!f,index:d2f00a00-fbf4-11ea-86fe-a5f65a6b0b79,key:log_processed.event.reason,negate:!f,params:(query:CREATE),type:phrase,value:CREATE),query:(match:(log_processed.event.reason:(query:CREATE,type:phrase)))),('$state':(store:appState),meta:(alias:!n,disabled:!f,index:d2f00a00-fbf4-11ea-86fe-a5f65a6b0b79,key:log_processed.event.source.component,negate:!f,params:(query:nginx-ingress-controller),type:phrase,value:nginx-ingress-controller),query:(match:(log_processed.event.source.component:(query:nginx-ingress-controller,type:phrase)))),('$state':(store:appState),meta:(alias:!n,disabled:!f,index:d2f00a00-fbf4-11ea-86fe-a5f65a6b0b79,key:log_processed.event.involvedObject.name,negate:!t,params:(query:integration-test-app-ing),type:phrase,value:integration-test-app-ing),query:(match:(log_processed.event.involvedObject.name:(query:integration-test-app-ing,type:phrase))))),index:d2f00a00-fbf4-11ea-86fe-a5f65a6b0b79,interval:auto,query:(language:kuery,query:''),sort:!(!('@timestamp',desc)))
```

This shows that there is a bulk create events, not sure yet what is causing it yet

**Comment by vijay-veeranki** [comment]
Drilling down to a particular namespace "poornima-staging" , as we know there is no pipeline running or any apply for this event.

https://kibana.cloud-platform.service.justice.gov.uk/_plugin/kibana/app/kibana#/discover?_g=(filters:!(),refreshInterval:(pause:!t,value:0),time:(from:'2020-09-29T16:00:00.000Z',to:'2020-09-29T17:00:00.000Z'))&_a=(columns:!(_source),filters:!(('$state':(store:appState),meta:(alias:!n,disabled:!f,index:d2f00a00-fbf4-11ea-86fe-a5f65a6b0b79,key:log_processed.event.involvedObject.kind,negate:!f,params:(query:Ingress),type:phrase,value:Ingress),query:(match:(log_processed.event.involvedObject.kind:(query:Ingress,type:phrase)))),('$state':(store:appState),meta:(alias:!n,disabled:!f,index:d2f00a00-fbf4-11ea-86fe-a5f65a6b0b79,key:log_processed.event.reason,negate:!f,params:(query:CREATE),type:phrase,value:CREATE),query:(match:(log_processed.event.reason:(query:CREATE,type:phrase)))),('$state':(store:appState),meta:(alias:!n,disabled:!f,index:d2f00a00-fbf4-11ea-86fe-a5f65a6b0b79,key:log_processed.event.source.component,negate:!f,params:(query:nginx-ingress-controller),type:phrase,value:nginx-ingress-controller),query:(match:(log_processed.event.source.component:(query:nginx-ingress-controller,type:phrase)))),('$state':(store:appState),meta:(alias:!n,disabled:!f,index:d2f00a00-fbf4-11ea-86fe-a5f65a6b0b79,key:log_processed.event.involvedObject.name,negate:!t,params:(query:integration-test-app-ing),type:phrase,value:integration-test-app-ing),query:(match:(log_processed.event.involvedObject.name:(query:integration-test-app-ing,type:phrase))))),index:d2f00a00-fbf4-11ea-86fe-a5f65a6b0b79,interval:auto,query:(language:kuery,query:%22poornima-staging%22),sort:!(!('@timestamp',desc)))



Same namespace for ingress-controller logs

https://kibana.cloud-platform.service.justice.gov.uk/_plugin/kibana/app/kibana#/discover?_g=(filters:!(),refreshInterval:(pause:!t,value:0),time:(from:'2020-09-29T16:00:00.000Z',to:'2020-09-29T17:00:00.000Z'))&_a=(columns:!(_source),filters:!(),index:'71644ed0-d648-11ea-b6f0-6bf964cd13a4',interval:auto,query:(language:kuery,query:%22poornima-staging%22),sort:!(!('@timestamp',desc)))

**Comment by vijay-veeranki** [comment]
https://kibana.cloud-platform.service.justice.gov.uk/_plugin/kibana/app/kibana#/discover?_g=(filters:!(),refreshInterval:(pause:!t,value:0),time:(from:'2020-09-29T16:01:10.540Z',to:'2020-09-29T16:02:10.560Z'))&_a=(columns:!(_source),filters:!(('$state':(store:appState),meta:(alias:!n,disabled:!f,index:d2f00a00-fbf4-11ea-86fe-a5f65a6b0b79,key:log_processed.event.message,negate:!f,params:(query:'Readiness%20probe%20failed:%20HTTP%20probe%20failed%20with%20statuscode:%20500'),type:phrase,value:'Readiness%20probe%20failed:%20HTTP%20probe%20failed%20with%20statuscode:%20500'),query:(match:(log_processed.event.message:(query:'Readiness%20probe%20failed:%20HTTP%20probe%20failed%20with%20statuscode:%20500',type:phrase))))),index:d2f00a00-fbf4-11ea-86fe-a5f65a6b0b79,interval:auto,query:(language:kuery,query:%22ingress-controller%22),sort:!(!('@timestamp',desc)))


Readiness probe failed: HTTP probe failed with statuscode: 500 for ingerss-controller at the same time the events got triggered
--

**Comment by vijay-veeranki** [comment]
{"verb":"UPDATED","event":{"metadata":{"name":"nginx-ingress-acme-controller-c9c8c7dd6-9qmxl.1638f4f7989b4df4","namespace":"ingress-controllers","selfLink":"/api/v1/namespaces/ingress-controllers/events/nginx-ingress-acme-controller-c9c8c7dd6-9qmxl.1638f4f7989b4df4","uid":"c4194bc9-1743-4611-8a6d-0e3ac1f47818","resourceVersion":"54538086","creationTimestamp":"2020-09-29T16:00:51Z"},"involvedObject":{"kind":"Pod","namespace":"ingress-controllers","name":"nginx-ingress-acme-controller-c9c8c7dd6-9qmxl","uid":"573acced-10c7-487a-a2c1-4bd7a5365788","apiVersion":"v1","resourceVersion":"382197390","fieldPath":"spec.containers{nginx-ingress-controller}"},"reason":"Unhealthy","message":"Readiness probe failed: HTTP probe failed with statuscode: 500","source":{"component":"kubelet","host":"ip-172-20-96-50.eu-west-2.compute.internal"},"firstTimestamp":"2020-09-28T13:16:51Z","lastTimestamp":"2020-09-29T16:01:21Z","count":5,"type":"Warning","eventTime":null,"reportingComponent":"","reportingInstance":""},"old_event":{"metadata":{"name":"nginx-ingress-acme-controller-c9c8c7dd6-9qmxl.1638f4f7989b4df4","namespace":"ingress-controllers","selfLink":"/api/v1/namespaces/ingress-controllers/events/nginx-ingress-acme-controller-c9c8c7dd6-9qmxl.1638f4f7989b4df4","uid":"c4194bc9-1743-4611-8a6d-0e3ac1f47818","resourceVersion":"54538062","creationTimestamp":"2020-09-29T16:00:51Z"},"involvedObject":{"kind":"Pod","namespace":"ingress-controllers","name":"nginx-ingress-acme-controller-c9c8c7dd6-9qmxl","uid":"573acced-10c7-487a-a2c1-4bd7a5365788","apiVersion":"v1","resourceVersion":"382197390","fieldPath":"spec.containers{nginx-ingress-controller}"},"reason":"Unhealthy","message":"Readiness probe failed: HTTP probe failed with statuscode: 500","source":{"component":"kubelet","host":"ip-172-20-96-50.eu-west-2.compute.internal"},"firstTimestamp":"2020-09-28T13:16:51Z","lastTimestamp":"2020-09-29T16:01:11Z","count":4,"type":"Warning","eventTime":null,"reportingComponent":"","reportingInstance":""}}

**Comment by vijay-veeranki** [comment]
Similar trigger of bulk events on 2020-09-29 05:35:21.709 "namespace":"poornima-staging"

https://kibana.cloud-platform.service.justice.gov.uk/_plugin/kibana/app/kibana#/discover?_g=(filters:!(),refreshInterval:(pause:!t,value:0),time:(from:'2020-09-01T16:00:00.000Z',to:'2020-09-29T17:00:00.000Z'))&_a=(columns:!(_source),filters:!(('$state':(store:appState),meta:(alias:!n,disabled:!f,index:d2f00a00-fbf4-11ea-86fe-a5f65a6b0b79,key:log_processed.event.involvedObject.kind,negate:!f,params:(query:Ingress),type:phrase,value:Ingress),query:(match:(log_processed.event.involvedObject.kind:(query:Ingress,type:phrase)))),('$state':(store:appState),meta:(alias:!n,disabled:!f,index:d2f00a00-fbf4-11ea-86fe-a5f65a6b0b79,key:log_processed.event.reason,negate:!f,params:(query:CREATE),type:phrase,value:CREATE),query:(match:(log_processed.event.reason:(query:CREATE,type:phrase)))),('$state':(store:appState),meta:(alias:!n,disabled:!f,index:d2f00a00-fbf4-11ea-86fe-a5f65a6b0b79,key:log_processed.event.source.component,negate:!f,params:(query:nginx-ingress-controller),type:phrase,value:nginx-ingress-controller),query:(match:(log_processed.event.source.component:(query:nginx-ingress-controller,type:phrase)))),('$state':(store:appState),meta:(alias:!n,disabled:!f,index:d2f00a00-fbf4-11ea-86fe-a5f65a6b0b79,key:log_processed.event.involvedObject.name,negate:!t,params:(query:integration-test-app-ing),type:phrase,value:integration-test-app-ing),query:(match:(log_processed.event.involvedObject.name:(query:integration-test-app-ing,type:phrase)))),('$state':(store:appState),meta:(alias:!n,disabled:!f,index:d2f00a00-fbf4-11ea-86fe-a5f65a6b0b79,key:kubernetes.container_name,negate:!f,params:(query:eventrouter),type:phrase,value:eventrouter),query:(match:(kubernetes.container_name:(query:eventrouter,type:phrase)))),('$state':(store:appState),meta:(alias:!n,disabled:!f,index:d2f00a00-fbf4-11ea-86fe-a5f65a6b0b79,key:log_processed.old_event.metadata.namespace,negate:!f,params:(query:poornima-staging),type:phrase,value:poornima-staging),query:(match:(log_processed.old_event.metadata.namespace:(query:poornima-staging,type:phrase)))),('$state':(store:appState),meta:(alias:!n,disabled:!f,index:d2f00a00-fbf4-11ea-86fe-a5f65a6b0b79,key:'@timestamp',negate:!f,params:(query:'2020-09-29T04:35:21.709Z'),type:phrase,value:'2020-09-29%2005:35:21.709'),query:(match:('@timestamp':(query:'2020-09-29T04:35:21.709Z',type:phrase))))),index:d2f00a00-fbf4-11ea-86fe-a5f65a6b0b79,interval:auto,query:(language:kuery,query:''),sort:!(!('@timestamp',desc)))


**Readiness probe failed: HTTP probe failed with statuscode: 500** at same time

https://kibana.cloud-platform.service.justice.gov.uk/_plugin/kibana/app/kibana#/discover?_g=(filters:!(),refreshInterval:(pause:!t,value:0),time:(from:'2020-09-29T04:35:15.000Z',to:'2020-09-29T04:36:00.000Z'))&_a=(columns:!(_source),filters:!(('$state':(store:appState),meta:(alias:!n,disabled:!f,index:d2f00a00-fbf4-11ea-86fe-a5f65a6b0b79,key:log_processed.event.metadata.namespace,negate:!f,params:(query:ingress-controllers),type:phrase,value:ingress-controllers),query:(match:(log_processed.event.metadata.namespace:(query:ingress-controllers,type:phrase)))),('$state':(store:appState),meta:(alias:!n,disabled:!f,index:d2f00a00-fbf4-11ea-86fe-a5f65a6b0b79,key:log_processed.event.involvedObject.name,negate:!f,params:(query:nginx-ingress-acme-controller),type:phrase,value:nginx-ingress-acme-controller),query:(match:(log_processed.event.involvedObject.name:(query:nginx-ingress-acme-controller,type:phrase)))),('$state':(store:appState),meta:(alias:!n,disabled:!f,index:d2f00a00-fbf4-11ea-86fe-a5f65a6b0b79,key:log_processed.event.message,negate:!f,params:(query:'Readiness%20probe%20failed:%20HTTP%20probe%20failed%20with%20statuscode:%20500'),type:phrase,value:'Readiness%20probe%20failed:%20HTTP%20probe%20failed%20with%20statuscode:%20500'),query:(match:(log_processed.event.message:(query:'Readiness%20probe%20failed:%20HTTP%20probe%20failed%20with%20statuscode:%20500',type:phrase))))),index:d2f00a00-fbf4-11ea-86fe-a5f65a6b0b79,interval:auto,query:(language:kuery,query:%22Readiness%22),sort:!(!('@timestamp',desc)))

**Comment by vijay-veeranki** [comment]
From the ingress-nginx guidence.

https://kubernetes.github.io/ingress-nginx/how-it-works/#when-a-reload-is-required


The next list describes the scenarios when a reload is required:

New Ingress Resource Created.
TLS section is added to existing Ingress.
Change in Ingress annotations that impacts more than just upstream configuration. For instance load-balance annotation does not require a reload.
A path is added/removed from an Ingress.
An Ingress, Service, Secret is removed.
Some missing referenced object from the Ingress is available, like a Service or Secret.
A Secret is updated.

**Comment by vijay-veeranki** [comment]
In live-1 cluster, we have new ingress getting created all the time as there are integration tests which create ingress for multiple tests. Also users create ingress to their services. Investigating to see if the relaod is the cause of this CREATE event.

**Comment by vijay-veeranki** [comment]
Reproduced the Ingress CREATE events in test cluster.

```docker-registry-cache   0s          Normal    CREATE                                                                                                      ingress/docker-registry-cache-ingress                        Ingress docker-registry-cache/docker-registry-cache-ingress
monitoring              0s          Normal    CREATE                                                                                                      ingress/prometheus-proxy-oauth2-proxy                        Ingress monitoring/prometheus-proxy-oauth2-proxy
kuberos                 0s          Normal    CREATE                                                                                                      ingress/kuberos                                              Ingress kuberos/kuberos
starter-pack            0s          Normal    CREATE                                                                                                      ingress/helloworld                                           Ingress starter-pack/helloworld
monitoring              0s          Normal    CREATE                                                                                                      ingress/prometheus-operator-grafana                          Ingress monitoring/prometheus-operator-grafana
monitoring              0s          Normal    CREATE                                                                                                      ingress/alertmanager-proxy-oauth2-proxy                      Ingress monitoring/alertmanager-proxy-oauth2-proxy
```

It is happening when node is recycled.

**Comment by vijay-veeranki** [comment]
We have node-recycler job which does terminate a node. This is triggering ingress CREATE event. This explains the reason why we see multiple CREATE events for Ingress

Events log related to ingress-controller when the node is recycled. 

```
default                 0s          Normal    NodeNotReady             node/ip-xxx-xx-xx-xxx.eu-west-2.compute.internal              Node ip-xxx-xx-xx-xxx.eu-west-2.compute.internal status is now: NodeNotReady
ingress-controllers     0s          Normal    UpdatedLoadBalancer      service/nginx-ingress-acme-controller                         Updated load balancer with new hosts
ingress-controllers     0s          Normal    UpdatedLoadBalancer      service/integration-test-nx-controller                        Updated load balancer with new hosts
default                 0s          Normal    Deleting node            node/ip-xxx-xx-xx-xxx.eu-west-2.compute.internal              Node ip-xxx-xx-xx-xxx.eu-west-2.comp
default                 0s          Normal    RemovingNode             node/ip-xxx-xx-xx-xxx.eu-west-2.compute.internal              Node ip-xxx-xx-xx-xxx.eu-west-2.compute.internal event: Removing Node ip-xxx-xx-xx-xxx.eu-west-2.compute.internal from Controller
ingress-controllers     0s          Normal    TaintManagerEviction     pod/nginx-ingress-acme-controller-c9c8c7dd6-vhtcc             Cancelling deletion of Pod ingress-controllers/nginx-ingress-acme-controller-c9c8c7dd6-vhtcc
ingress-controllers     0s          Normal    Scheduled                pod/nginx-ingress-acme-controller-c9c8c7dd6-7cl4t             Successfully assigned ingress-controllers/nginx-ingress-acme-controller-c9c8c7dd6-7cl4t to ip-xxx-xx-xx-xxx.eu-west-2.compute.internal
ingress-controllers     0s          Normal    SuccessfulCreate         replicaset/nginx-ingress-acme-controller-c9c8c7dd6            Created pod: nginx-ingress-acme-controller-c9c8c7dd6-7cl4t
ingress-controllers     0s          Warning   FailedMount              pod/nginx-ingress-acme-controller-c9c8c7dd6-7cl4t             MountVolume.SetUp failed for volume "nginx-ingress-acme-token-sgjmx" : failed to sync secret cache: timed out waiting for the condition
ingress-controllers     0s          Warning   FailedMount              pod/nginx-ingress-acme-controller-c9c8c7dd6-7cl4t             MountVolume.SetUp failed for volume "webhook-cert" : failed to sync secret cache: timed out waiting for the condition
ingress-controllers     0s          Warning   FailedMount              pod/nginx-ingress-acme-controller-c9c8c7dd6-7cl4t             MountVolume.SetUp failed for volume "nginx-ingress-acme-token-sgjmx" : failed to sync secret cache: timed out waiting for the condition
ingress-controllers     0s          Normal    Pulling                  pod/nginx-ingress-acme-controller-c9c8c7dd6-7cl4t             Pulling image "quay.io/kubernetes-ingress-controller/nginx-ingress-controller:0.30.0"
ingress-controllers     61m         Normal    Scheduled                pod/nginx-ingress-acme-controller-c9c8c7dd6-b2mj5             Successfully assigned ingress-controllers/nginx-ingress-acme-controller-c9c8c7dd6-b2mj5 to ip-xxx-xx-xx-xxx.eu-west-2.compute.internal
ingress-controllers     60m         Normal    Pulling                  pod/nginx-ingress-acme-controller-c9c8c7dd6-b2mj5             Pulling image "quay.io/kubernetes-ingress-controller/nginx-ingress-controller:0.30.0"
ingress-controllers     60m         Normal    Pulled                   pod/nginx-ingress-acme-controller-c9c8c7dd6-b2mj5             Successfully pulled image "quay.io/kubernetes-ingress-controller/nginx-ingress-controller:0.30.0"
ingress-controllers     60m         Normal    Created                  pod/nginx-ingress-acme-controller-c9c8c7dd6-b2mj5             Created container nginx-ingress-controller
ingress-controllers     60m         Normal    Started                  pod/nginx-ingress-acme-controller-c9c8c7dd6-b2mj5             Started container nginx-ingress-controller
ingress-controllers     61m         Normal    SuccessfulCreate         replicaset/nginx-ingress-acme-controller-c9c8c7dd6            Created pod: nginx-ingress-acme-controller-c9c8c7dd6-b2mj5
ingress-controllers     61m         Normal    Killing                  pod/nginx-ingress-acme-controller-c9c8c7dd6-lcksf             Stopping container nginx-ingress-controller
ingress-controllers     60m         Warning   Unhealthy                pod/nginx-ingress-acme-controller-c9c8c7dd6-lcksf             Liveness probe failed: HTTP probe failed with statuscode: 500
ingress-controllers     60m         Warning   Unhealthy                pod/nginx-ingress-acme-controller-c9c8c7dd6-lcksf             Readiness probe failed: HTTP probe failed with statuscode: 500
ingress-controllers     61m         Normal    CREATE                   configmap/nginx-ingress-acme-controller                       ConfigMap ingress-controllers/nginx-ingress-acme-controller
kuberos                 60m         Normal    CREATE                   ingress/kuberos                                               Ingress kuberos/kuberos
monitoring              60m         Normal    CREATE                   ingress/alertmanager-proxy-oauth2-proxy                       Ingress monitoring/alertmanager-proxy-oauth2-proxy
monitoring              60m         Normal    CREATE                   ingress/prometheus-operator-grafana                           Ingress monitoring/prometheus-operator-grafana
monitoring              60m         Normal    CREATE                   ingress/prometheus-proxy-oauth2-proxy                         Ingress monitoring/prometheus-proxy-oauth2-proxy
starter-pack            60m         Normal    CREATE                   ingress/helloworld                                            Ingress starter-pack/helloworld
vijay-test              60m         Normal    CREATE                   ingress/helloworld-rubyapp-ingress2                           Ingress vijay-test/helloworld-rubyapp-ingress2
ingress-controllers     0s          Normal    Pulled                   pod/nginx-ingress-acme-controller-c9c8c7dd6-7cl4t             Successfully pulled image "quay.io/kubernetes-ingress-controller/nginx-ingress-controller:0.30.0"
ingress-controllers     0s          Normal    Created                  pod/nginx-ingress-acme-controller-c9c8c7dd6-7cl4t             Created container nginx-ingress-controller
ingress-controllers     0s          Normal    Started                  pod/nginx-ingress-acme-controller-c9c8c7dd6-7cl4t             Started container nginx-ingress-controller
ingress-controllers     0s          Normal    CREATE                   configmap/nginx-ingress-acme-controller                       ConfigMap ingress-controllers/nginx-ingress-acme-controller
monitoring              0s          Normal    CREATE                   ingress/alertmanager-proxy-oauth2-proxy                       Ingress monitoring/alertmanager-proxy-oauth2-proxy
kuberos                 0s          Normal    CREATE                   ingress/kuberos                                               Ingress kuberos/kuberos
monitoring              0s          Normal    CREATE                   ingress/prometheus-operator-grafana                           Ingress monitoring/prometheus-operator-grafana
starter-pack            0s          Normal    CREATE                   ingress/helloworld                                            Ingress starter-pack/helloworld
monitoring              0s          Normal    CREATE                   ingress/prometheus-proxy-oauth2-proxy                         Ingress monitoring/prometheus-proxy-oauth2-proxy
vijay-test              0s          Normal    CREATE                   ingress/helloworld-rubyapp-ingress2                           Ingress vijay-test/helloworld-rubyapp-ingress2
docker-registry-cache   0s          Normal    CREATE                   ingress/docker-registry-cache-ingress                         Ingress docker-registry-cache/docker-registry-cache-ingress
ingress-controllers     0s          Normal    UpdatedLoadBalancer      service/nginx-ingress-acme-controller                         Updated load balancer with new hosts
ingress-controllers     0s          Normal    UpdatedLoadBalancer      service/integration-test-nx-controller                        Updated load balancer with new hosts
```

**Comment by digitalronin** [comment]
Do we know why creating ingresses causes problems? Recycling nodes is something we should do (and are doing) on a regular basis. If we ever move to using spot instances for some worker nodes, it's going to happen even more often.

So, it's great that we know what's causing the ingress create events, but I'm also interested in whether that's really *causing* problems and, if so, why?

**Comment by vijay-veeranki** [comment]
Regarding this
"So, it's great that we know what's causing the ingress create events.

```
ingress-controllers     0s          Normal    UpdatedLoadBalancer      service/nginx-ingress-acme-controller                         Updated load balancer with new hosts
ingress-controllers     0s          Normal    UpdatedLoadBalancer      service/integration-test-nx-controller                        Updated load balancer with new hosts
```

This two events here, might be the cause of it, I will raise this as an issue with ingress-controller to see if other users noticed it.

Regarding:

"but I'm also interested in whether that's really causing problems and, if so, why?"

We see this CREATE ingress events for all the ingress including our services, we didn't notice any alerts when this event get triggered.

**Comment by vijay-veeranki** [comment]
Raised this to see if any other users seen this behaviour.

https://github.com/kubernetes/ingress-nginx/issues/6279

https://kubernetes.slack.com/archives/CANQGM8BA/p1601899125390600

**Comment by vijay-veeranki** [comment]
Got a reply for the issue raised.

https://github.com/kubernetes/ingress-nginx/issues/6279
This is causing confusion as the actual ingress is not created, but this event is created.

The CREATE is related to the ingress controller, not the ingress definition. That said, I see this could be confusing because of the lack of context.

**Comment by jasonBirchall** [comment]
So the CREATE event is the ingress controller pod recycling?

**Comment by poornima-krishnasamy** [comment]
It looks the CREATE event is happening whenever the nginx pods are recycled. When the ingress-controller pods are recycled, it sends an event to UpdateLoadBalancer. Possibly to say that there is a new node which has a healthy target of nginx pods. 

https://kibana.cloud-platform.service.justice.gov.uk/_plugin/kibana/app/kibana#/discover?_g=(filters:!(),refreshInterval:(pause:!t,value:0),time:(from:'2020-12-07T00:00:00.000Z',to:now))&_a=(columns:!(_source),filters:!(('$state':(store:appState),meta:(alias:!n,disabled:!f,index:d2f00a00-fbf4-11ea-86fe-a5f65a6b0b79,key:log_processed.event.involvedObject.kind,negate:!f,params:(query:Ingress),type:phrase,value:Ingress),query:(match:(log_processed.event.involvedObject.kind:(query:Ingress,type:phrase)))),('$state':(store:appState),meta:(alias:!n,disabled:!f,index:d2f00a00-fbf4-11ea-86fe-a5f65a6b0b79,key:log_processed.event.reason,negate:!f,params:(query:CREATE),type:phrase,value:CREATE),query:(match:(log_processed.event.reason:(query:CREATE,type:phrase)))),('$state':(store:appState),meta:(alias:!n,disabled:!f,index:d2f00a00-fbf4-11ea-86fe-a5f65a6b0b79,key:log_processed.event.source.component,negate:!f,params:(query:nginx-ingress-controller),type:phrase,value:nginx-ingress-controller),query:(match:(log_processed.event.source.component:(query:nginx-ingress-controller,type:phrase))))),index:d2f00a00-fbf4-11ea-86fe-a5f65a6b0b79,interval:m,query:(language:kuery,query:''),sort:!(!('@timestamp',desc)))

**Comment by aledbf** [comment]
This issue is fixed in 0.41.0 https://github.com/kubernetes/ingress-nginx/pull/6375

**Comment by AntonyBishop** [comment]
We need to check fix about addresses this issue.
